primo-explore/custom/01MIT_INST-TACOS/js/custom.js

```diff
@@ -14,8 +42,56 @@
       '<div id="title-mit"><a href="https://mit.primo.exlibrisgroup.com/discovery/search?vid=01MIT_INST:MIT&lang=en">Search Our Collections</a></div>',
   });
   app.component("prmSearchBarAfter", {
-    template: alertBannerTemplate,
+    template: `<tacos></tacos>${alertBannerTemplate}`,
+  });
+  app.component("tacos", {
+    controller: function tacosController($scope, tacosService, $stateParams) {
+      // $stateParams seems to behave like an angularJS service, but I'm not sure
+      // where in the Primo VE source code it is defined...
+      // We are able to inject it in the controller and then get access to the .query property
+      // which is the raw search string(s) we want to send to TACOS
+
+      var vm = this;
+
+      function parseSearchQuery(searchQuery) {
+        // Returns searchQuery if it is a string.
+        // If searchQuery is an array of strings, joins the strings and returns the result
+
+        if (typeof searchQuery === "string") { // handle a string (simple search)
+          return searchQuery;
+        }
+        return searchQuery.join(" "); // handle an array of strings (advanced search)
+      }
+
+      $scope.$watch(function () {
+        // Watch for changes to the $stateParams.query property
+
+        return parseSearchQuery(
+          $stateParams.query,
+        );
+      }, function (newSearchQuery) {
+        // This listener function triggers when the watcher is first initialized and
+        // each time the results of the watched expression change afterward.

```

Comment:
I'm curious to see what sort of volume this approach generates within TACOS. Am I correct in thinking that users can end up conducting many searches - at times with little relationship to each other - within the context of a single page load? Or are these subsequent triggers more akin to search refinements rather than starting a new search from scratch?

Reply:
This function would run whenever a search is submitted where there is a change to either the search terms, the search field (`any `vs. `title`), and the precision (`contains` vs. `begins_with`). other query string parameters (filters, pagination) wouldn't count as a change.

Reply:
Ok - when I was poking around, I think I saw a split like this but I wasn't sure what the exact threshold was. 

To be clear, this isn't a reason to change anything in this PR - but I'm anticipating that we'll see what the data looks like and then determine whether any adjustments might be needed.